app/models/embedded_ansible_worker/runner.rb

```diff
@@ -22,6 +23,7 @@ def heartbeat
   end
 
   def do_work
+    synchronize_provider_with_server unless provider_in_sync_with_server?
```

Comment:
I think this new method name looks better... @carbonin what do you think?

Reply:
I think the same problem is still there.

The name only mentions that we're checking the properties, not that we are also returning true based on some timeframe.  It doesn't really matter though. I'm not going to hold this up for a better name though.